demo/src/main/java/com/novoda/accessibility/demo/custom_actions/TweetView.java

```diff
@@ -44,23 +43,43 @@ protected void onFinishInflate() {
     }
 
     public void display(String tweet, Listener listener) {
-        Actions actions = createActions(tweet, listener);
-        ActionsAccessibilityDelegate delegate = new ActionsAccessibilityDelegate(getResources(), actions);
-        delegate.setClickLabel(R.string.tweet_actions_usage_hint);
+        MenuItem.OnMenuItemClickListener menuItemClickListener = createMenuItemClickListener(tweet, listener);
+        Menu menu = actionsMenuInflater.inflateMenu(R.menu.tweet_actions, menuItemClickListener);
+        ActionsMenuAccessibilityDelegate delegate = new ActionsMenuAccessibilityDelegate(menu, menuItemClickListener);
+//      TODO: add composable delegate - delegate.setClickLabel(R.string.tweet_actions_usage_hint);
```

Comment:
not sure about this bit yet - I was going to add `UsageHintsAccessibilityDelegate` like in `ActionsAccessibilityDelegate` but it doesn't seem very scaleable.

I wonder if it's better to take a collection of `AccessibilityDelegate` and have a composite or some sort.

Reply:
On looking at this, I'm not sure it's possible to do this - at least it's not immediately obvious how a CompositeAccessibilityDelegate would behave.

There are many methods inside AccessibilityDelegate and it's unclear whether it's appropriate to call _each_ of the delegates or just first one that accepts the event (some methods return boolean).

For this reason, I think it'd be most appropriate if we either let the user define their own composite delegate, in which they can call to the UsageHints one and the `ActionsMenuAccessibilityDelegate` OR add `UsageHints` behaviour to the `Actions` one as we have already in `ActionsAccessibilityDelegate`.

I'd recommend the former for now, since we'll have the chance to add it later, whereas adding it now will mean we might have to remove it later which is more pain.